src/IssueViz/IssueVisualizationControl/IssueVisualizationControl.xaml.cs

```diff
@@ -18,17 +18,39 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
 using System.Windows.Controls;
+using System.Windows.Navigation;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.IssueVisualization.IssueVisualizationControl.ViewModels;
 
 namespace SonarLint.VisualStudio.IssueVisualization.IssueVisualizationControl
 {
-    internal sealed partial class IssueVisualizationControl : UserControl
+    internal sealed partial class IssueVisualizationControl : UserControl, IDisposable
     {
-        public object ViewModel { get; }
-        
-        public IssueVisualizationControl()
+        public IssueVisualizationViewModel ViewModel { get; }
+
+        public IssueVisualizationControl(IssueVisualizationViewModel viewModel)
         {
+            ViewModel = viewModel;
+
             InitializeComponent();
         }
+
+        public void Dispose()
+        {
+            ViewModel?.Dispose();
```

Comment:
It seems odd that the control is disposing something it didn't create. Both the view model and the control are created by the `IssueVisualizationToolWindow`, which already implements `IDisposable`. That would seem like a more natural place to dispose the view model.
In practical terms it doesn't look like it will make much difference since the lifetimes of the window pane and the control appear to be linked.